docs/contributors/code/release.md

```diff
@@ -48,16 +48,28 @@ You should check that folks are able to install the new version from their Dashb
 
 Once released, all that's left to do is writing a release post on [make.wordpress.org/core](https://make.wordpress.org/core/). You can find some tips on that below.
 
+#### Troubleshooting
+
+The workflow is quite resilient, but it is possible that a stage may sometimes fail leading to a piece of the release process not completing correctly.
+
+In this scenario it is usually easiest to simply revert and start over. To do this:
+
+1. Revert any version bump commits on `trunk` and then push the revert commit directly to `trunk`. This will require `admin` level privillages on the repo, so you may need to ask a member of the Core team for assistance.
+2. Delete the `release` _branch_ on the remote - `git push origin --delete release/X.Y.Z`
```

Comment:
This only applies to RCs (as that's when a `release` branch is created) 🤔  We shouldn't really be deleting them at a later stage (i.e. stable or point release), as we'd lose information about which commits are included in the branch...

So we'd need to nuance this a bit more. That might mean that instructions will become a bit more confusing though... 🙁

Reply:
We can suggest reverting the version bump commit on the release branch (and pushing the updated branch), or to delete the branch if we're trying to release an RC. 

Or maybe I should just try to make the flow more resilient and e.g. ignore a version bump commit if it'd be a no-op 🤔

Reply:
I had another look at the original issue:

- On first try, [the plugin build step failed](https://github.com/WordPress/gutenberg/runs/2951792952?check_suite_focus=true), apparently due to some npm flakiness. It did however create the `release/11.0` branch, and pushed the version bump commit to it.
- On second try, [the version bump step failed](https://github.com/WordPress/gutenberg/runs/2952331107?check_suite_focus=true). This makes sense: It successfully created the `release/11.0` branch locally and committed the version bump to it, but failed when trying to push it to the remote. However, there was already a `release/11.0` remote branch from the first attempt.

Reply:
In order to safeguard against this kind of scenario, we could unconditionally create the `release` branch, and override any potentially existing such branch. However, this only works for RCs.

If we also want to make the stable release process more resilient, things get quite a bit trickier, since there are a lot of edge cases 😕  Generally, we'd want to keep the `release` branch in this scenario, since it probably contains a history of commits (and maybe point release version bumps) that we'd like to retain. 

While we could add a check that only bumps the version if it hasn't been bumped to the desired version yet, such a check becomes unreliable if the workflow is run from a previous release branch (in order to create a new point release in that series). Overall, things seem to get complex (and potentially messy) quickly in this scenario.

A more promising strategy would be for the workflow to self-heal, i.e. if the build step doesn't succeed, we revert the version bump commit. While GHA have [job status check functions](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#job-status-check-functions), I don't think they work across jobs -- but I'd rather not do the (conditional) revert in the same job as the build... 🤔

Reply:
Here's a draft PR: #33239